src/app/preference/preference-modal/preference-modal.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, OnInit, Output, EventEmitter, OnDestroy } from '@angular/core';
+import { ModalController } from '@ionic/angular';
+import { PreferenceService, Category } from '@services/preference.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subscription } from 'rxjs/Subscription';
+import { UtilsService } from '@services/utils.service';
+import { ThrowStmt } from '@angular/compiler';
+@Component({
+  selector: 'app-preference-modal',
+  templateUrl: './preference-modal.component.html',
+  styleUrls: ['./preference-modal.component.scss']
+})
+export class PreferenceModalComponent implements OnDestroy, OnInit {
+  preferences$ = this.preferenceService.preference$;
+  preferenceSubject$: Subscription;
+  currentPreference;
+  showUpdateModal = false;
+  private key: string;
+  preferences: {
+    categories: any;
+  };
+
+  constructor(
+    private preferenceService: PreferenceService,
+    public modalController: ModalController,
+    private activatedRoute: ActivatedRoute,
+    private router: Router
+  ) {
+  }
+  ngOnInit() {
+    this.preferenceSubject$ = this.preferenceService.getPreference();
+    this.currentPreference = {
+      name: '',
+      description: '',
+      options: '',
+      remarks: '',
+    };
+  }
+  close () {
+    this.modalController.dismiss();
+  }
+  PreferenceUpdateModal(event, key) {
```

Comment:
the naming convention should follow [camelCase](https://www.w3schools.com/js/js_conventions.asp#:~:text=camelCase%3A,with%20many%20JavaScript%20library%20names.)